drivers/driver.go

```diff
@@ -337,7 +338,14 @@ func New(name string, config Options) (Driver, error) {
 
 	// Guess for prior driver
 	driversMap := scanPriorDrivers(config.Root)
-	for _, name := range priority {
+
+	// use the supplied priority list unless it is empty
+	prioList := config.DriverPriority
+	if len(prioList) == 0 {
+		prioList = priority
+	}
+
+	for _, name := range prioList {
 		if name == "vfs" {
```

Comment:
if vfs is set in the `config.DriverPriority` then I expect it to be honored.

I think it should be changed to `name == "vfs" && len(config.DriverPriority) == 0`